tests/object_test.py

```diff
@@ -188,6 +188,25 @@ def __init__(self):
         # Keep track of how many times this constructor has been run.
         ThingWithDefaultFactoryRegistry.count += 1
 
+class ThingWithIteratorPlusData:
+    """ Class that has an iterator and more"""
+    def __init__(self, a):
+        self.a = a
+        self._data = [3,1,4,1,5,9]
+        self._index = 0
+
+    def __iter__(self):
+        self._index = 0
+        return self
+
+    def __next__(self):
+        try:
+            item = self._data[self._index]
+        except IndexError as e:
+            raise StopIteration from e
```

Comment:
Is the ``from e`` syntax compatible with Python versions as low as 3.7? I could have sworn that ``raise Exception from e`` was a newer construct, but I may be wrong.

Reply:
It seems the tests passed on 3.7, so I guess this is all good.